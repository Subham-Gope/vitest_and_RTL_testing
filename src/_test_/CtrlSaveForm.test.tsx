// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import CtrlSaveForm from "../components/CtrlSaveForm";

// describe("Testing the CtrlSaveForm.tsx file", () => {
//   test("saves the file on ctrl + s", async () => {
//     // setting up the userEvent
//     const user = userEvent.setup();

//     const { container } = render(<CtrlSaveForm />);
//     expect(container).toMatchSnapshot();

//     const input = screen.getByLabelText(/enter your/i) as HTMLInputElement;
//     expect(input).toBeInTheDocument();

//     expect(input.value).toBe("");

//     // intiating user interaction
//     input.focus();
//     await user.type(input, "Hello");
//     await user.keyboard("[CtrlLeft>][KeyS][/CtrlLeft]");

//     const text = screen.getByText(/hello/i);
//     expect(text).toBeInTheDocument();
//   });
// });
