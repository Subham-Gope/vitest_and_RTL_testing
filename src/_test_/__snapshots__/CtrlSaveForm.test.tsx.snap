// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the CtrlSaveForm.tsx file > saves the file on ctrl + s 1`] = `
<div>
  <div>
    <label
      for="ctrl-s-input"
    >
      Enter Your Text
    </label>
    <input
      id="ctrl-s-input"
      placeholder="Example Text"
      type="text"
      value=""
    />
  </div>
</div>
`;
